dm/common/common.go

```diff
@@ -173,3 +188,113 @@ func (s keyHexEncoderDecoder) Decode(key string) ([]string, error) {
 func (s keyHexEncoderDecoder) Path() string {
 	return string(s)
 }
+
+// version V2 fixed the bug that XXDecoder.Encode("s1") is the prefix of XXDecoder.Encode("s10"), so watching prefix
+// gets unexpected KV
+// also, always use keyHexEncoderDecoderV2 to avoid `/` in keys
+type keyHexEncoderDecoderV2 string
+
+func quotes(s string) string {
+	return "`" + s + "`"
```

Comment:
Why don't we just add a separator to the end of the encoded key when watching prefix? It can simply avoid watching incorrect keys.

Reply:
do you mean s + `sep`? I just think quoting something is more familiar to others like an identifier in SQL, while familiar separators such as \0, ctrl+d, are not easy to type

Reply:
Yes. It seems all of the `KeyAdapter`s encode the custom keys to hex and joins them with '/'.  I'm not sure whether I have understood the scenario of watching prefix. If we want to watch the prefix `/dm/s1`, the expected keys should be `/dm/s1/xxx`? If so, adding '/' to the end of key looks more reasonable.

Reply:
Oh I see, will check it later

Reply:
👍 I'm going to add an `EncodeAsPrefix`, which checks and returns a `/` terminated string

Reply:
still need to check watch `/dm/s1\0` could receive changes of `/dm/s1`, because there maybe `/dm/s10` 🤔

Reply:
does we actually want to watch `/dm/s1` instead of `/dm/s1/xxx`,  in this case,  just watch the key only.